lib/Modules/FreeRdpHooks/FreeRdpHooks.psm1

```diff
@@ -30,11 +30,19 @@ function Get-Vcredist {
         try {
             Write-JujuWarning "Trying to get vcredist Juju resource"
             $vcredistPath = Get-JujuResource -Resource "vcredist-x64"
-            return $vcredistPath
         } catch {
+            $vcredistPath = $null
             Write-JujuWarning "Failed downloading vcredist resource: $_"
-            Write-JujuWarning "Falling back to file download"
         }
+        if ($vcredistPath) {
+            $bytes = Get-Content $vcredistPath -TotalCount 31 -Encoding Byte
+            $str = [string]::join("", [char[]]$bytes)
+            if ($str -eq $DEFAULT_JUJU_RESOURCE_CONTENT) {
+                Throw "Manually attach the Juju resource 'vcredist-x64' or set the config option 'vcredist-url'."
```

Comment:
I would rather we

```powershell
Set-JujuStatus -Status blocked -Message "You myst upload vcredist-x64 resource"
```

and return then ```Throw```. Otherwise the user will only see an error status with no other meaningful information, unless he looks at the log.